katharsis-core/src/main/java/io/katharsis/resource/registry/responseRepository/ResourceRepositoryAdapter.java

```diff
@@ -1,109 +1,189 @@
-		}
-		else {
-			((ResourceRepository) resourceRepository).delete(id);
-		}
-		return new JsonApiResponse();
-	}
-
-	public Object getResourceRepository() {
-		return resourceRepository;
-	}
-
-	@Override
-	protected Class<?> getResourceClass(Object repository) {
-		return resourceInformation.getResourceClass();
-	}
-}
+package io.katharsis.resource.registry.responseRepository;
+
+import java.io.Serializable;
+
+import io.katharsis.dispatcher.controller.HttpMethod;
+import io.katharsis.module.ModuleRegistry;
+import io.katharsis.queryspec.QuerySpecResourceRepository;
+import io.katharsis.queryspec.internal.QueryAdapter;
+import io.katharsis.repository.ResourceRepository;
+import io.katharsis.repository.annotated.AnnotatedResourceRepositoryAdapter;
+import io.katharsis.repository.filter.RepositoryFilterContext;
+import io.katharsis.request.repository.RepositoryRequestSpec;
+import io.katharsis.resource.information.ResourceInformation;
+import io.katharsis.response.JsonApiResponse;
+
+/**
+ * A repository adapter for resource repository.
+ */
+@SuppressWarnings("unchecked")
+public class ResourceRepositoryAdapter<T, I extends Serializable> extends ResponseRepository {
+
+	private final Object resourceRepository;
+
+	private final boolean isAnnotated;
+
+	public ResourceRepositoryAdapter(ResourceInformation resourceInformation, ModuleRegistry moduleRegistry,
+			Object resourceRepository) {
+		super(resourceInformation, moduleRegistry);
+		this.resourceRepository = resourceRepository;
+		this.isAnnotated = resourceRepository instanceof AnnotatedResourceRepositoryAdapter;
+	}
+
+	public JsonApiResponse findOne(I id, QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+				Serializable id = request.getId();
+				Object resource;
+				if (isAnnotated) {
+					resource = ((AnnotatedResourceRepositoryAdapter) resourceRepository).findOne(id, queryAdapter);
+				}
+				else if (resourceRepository instanceof QuerySpecResourceRepository) {
+					resource = ((QuerySpecResourceRepository) resourceRepository).findOne(id,
+							request.getQuerySpec(resourceInformation.getResourceClass()));
+				}
+				else {
+					resource = ((ResourceRepository) resourceRepository).findOne(id, request.getQueryParams());
+				}
+				return getResponse(resourceRepository, resource, request);
+			}
+
+		};
+		RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forFindId(moduleRegistry.getResourceRegistry(),
+				queryAdapter, id);
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+	}
+
+	public JsonApiResponse findAll(QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)